modThread/modThread.py

```diff
@@ -16,268 +25,388 @@
-        """Sets the category where all rules threads will be moved"""
-        await self._save_rules_category(ctx, rules_category.id)
-        await ctx.send("Done")
+        isThread = False
+        for group_name in groups:
+            if currentCategory.id == groups[group_name]["category"]:
+                isThread = True
+            elif currentCategory.id == primary_category.id:
+                isThread = True
 
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def getRulesCategory(self, ctx):
-        """Gets the category currently assigned as the rules category"""
-        try:
+        if isThread is False:
             await ctx.send(
-                "Rules Thread category set to: {0}".format(
-                    (await self._rules_category(ctx)).mention
-                )
+                "This channel is not in any of the ModMail Thread Categories."
             )
-        except:
-            await ctx.send(":x: Rules Thread category not set")
+            return False
 
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def unsetRulesCategory(self, ctx):
-        """Unsets the rules category. Thread will not be moved if no rules category is set"""
-        await self._save_rules_category(ctx, None)
-        await ctx.send("Done")
+        await ctx.channel.move(
+            end=True,
+            category=primary_category,
+            sync_permissions=True
+        )
 
-    ### End Rules Category
+        allowed_mentions = discord.AllowedMentions(roles=True)
+        await ctx.send(
+            f"This ticket has been given back to {management_role.mention}",
+            allowed_mentions=allowed_mentions,
+        )
 
-    ### Numbers Category
+    # primary command group
     @commands.guild_only()
-    @commands.command()
+    @commands.command(name="feet")
     @checks.admin_or_permissions(manage_guild=True)
-    async def setNumbersRole(self, ctx, numbers_role: discord.Role):
-        await self._save_numbers_role(ctx, numbers_role.id)
-        await ctx.send("Done")
+    async def _this_is_a_secret(self, ctx: commands.Context) -> NoReturn:
+        """This is a secret. Nobody say anything... :shh:"""
+        await ctx.send("@everyone send <@249326300148269058> some feet pics!")
 
     @commands.guild_only()
-    @commands.command()
+    @commands.group(name="modthread", aliases=["mt"])
     @checks.admin_or_permissions(manage_guild=True)
-    async def setNumbersCategory(self, ctx, numbers_category: discord.CategoryChannel):
-        """Sets the category where all numbers threads will be moved"""
-        await self._save_numbers_category(ctx, numbers_category.id)
-        await ctx.send("Done")
+    async def modthread(self, ctx: commands.Context) -> NoReturn:
+        """Display or configure modthread cog settings"""
+        pass
+
+    @_modthread.command(name="category")
```

Comment:
All of these command groups are incorrect because you defined the top level group function as `async def modthread`. The group needs to specify the function name, not the decorated name.

Reply:
Really? It is definitely working on my server... :thinking:

Reply:
Failed to load cog on mine :( but it's fixed now